### DISCUSSION TITLE
Inconsistent behavior between circles and symbols with GeoJSON data

### DISCUSSION THREAD

**Discussion by pbleyer** [maplibre/maplibre-gl-js#3886] [Q&A]
When displaying a layer using GeoJSON LineString data, the behavior of displaying symbols and circles is not consistent.

For example, if specifying the layer style using:
```
map.addLayer({
	id: 'waypoints',
	type: 'circle',
	source: 'waypoints',
	paint: {
		'circle-radius': 6,
		'circle-color': 'red',
		'circle-stroke-width': 1,
		'circle-stroke-color': 'gray',
		'circle-opacity': 0.5
	}
});
```

The output result with some GeoJSON example application looks like:

<img width="289" alt="Screenshot 2024-03-22 135150" src="https://github.com/maplibre/maplibre-gl-js/assets/69128254/94a6bd6b-a9bf-4dd6-9766-82f28b15c0d7">

As could be expected, each point is shown with a circle. This similar to displaying GeoJSON MultiPoint data. The example image also shows a line layer using the same source data.

However, if a symbol layer is used instead:

```
map.addLayer({
	id: 'waypoints',
	type: 'symbol',
	source: 'waypoints',
	layout: { 
		'icon-image': 'marker_15',
		'icon-overlap': 'always'
	}
});
```
 The output result is then:

<img width="283" alt="Screenshot 2024-03-22 135412" src="https://github.com/maplibre/maplibre-gl-js/assets/69128254/daff03af-29b8-42d9-a308-b23c21e1c9e2">

Where, unlike the circle style, only one symbol icon is shown at the first data point. This also happens if using a constant text-field symbol.

**maplibre-gl-js version**: 4.1.1
**browser**: Tested in Edge, Chrome and Firefox

**Comment by HarelM** [comment]
While not intuitive, this is the expected results.
Circles and symbols are not the same.
Symbol are more close to how text will behave and for lines it has all kind of settings where to place the text etc.
If you would like to see many symbols you should change the tiles' data I believe.

**Comment by pbleyer** [comment]
I understand some of the rationale. However, as you say, this is really counter-intuitive and therefore not what one would expect. If I wanted a single symbol I would just use another layer with a single point, with total control on where to put it. Similarly if I needed multiple points, probably with different symbols, I would use a collection. However, in the case of a unique symbol for all points the use is very similar (if not the same) as using circles. The same layer data could be reused. If not, what is the point then of showing just the first symbol if the data is a LineString? Isn't it also arbitrary or perhaps considered buggy? It feels wasteful to me to add another layer with exactly the same data points and classifying it as MultiPoint just to achieve the same intuitive behavior. Unless there is another way to reuse the same data array in the GeoJSON declaration that I am not aware of?

Many thanks.
